packages/mui-material/src/Chip/Chip.js

```diff
@@ -152,7 +152,11 @@ const ChipRoot = styled('div', {
           marginLeft: -4,
         }),
         ...(ownerState.color !== 'default' && {
-          color: alpha(theme.palette[ownerState.color].contrastText, 0.7),
+          color: alpha(
```

Comment:
If I am not mistaken by what you mean and as I said in the description, the `contrastText` is already added for `primary`, `secondary` etc color palettes. This fix is if custom color palettes are provided (Here the key is `custom`).

eg:
```jsx
palette: {
  custom: {
    main: '#f90',
  },
},
```

Reply:
Yes, I was wondering if we should do the same for the custom colors, but it may be overkill, basically to invoke `augmentColor` for the custom colors too - https://github.com/mui/material-ui/blob/master/packages/mui-material/src/styles/createPalette.js#L279 

I am starting this conversation, as although the fix works at this moment, when we use CSS vars we would hit this problem again.

Reply:
@siriwatknp What do you think?

Reply:
I think the same. We should expose the `createPalette` if developers want to add extra palettes to be used with any `<Component color="..." />`. This way it ensures consistency and we don't have to worry about the implementation of the component.

Reply:
> We should expose the `createPalette`

Can you guide me by what you have in mind in how to do this?

Reply:
Also if we do what you'll are saying, I think it would be a breaking change.

`contrastText` will be applied here [in this type of case](https://github.com/mui/material-ui/blob/32292cbd74427f6a4ea3b379df697913ad79b86a/packages/mui-material/src/Chip/Chip.js#L166) or similarly elsewhere since it isn't working for custom color palettes.

My PR is just a fix within alpha so that the component does not crash.

Reply:
I am leaving the decision on @siriwatknp, I've already left my opinion on the matter :)

Reply:
@ZeeshanTamboli What if we starts with docs first about adding new color to the palette so that it works with `color="new-color"` prop?

Reply:
@siriwatknp I didn't understand what you mean. Is it related to this PR's objective? We already have the [adding new colors](https://mui.com/material-ui/customization/palette/#adding-new-colors) documentation.

Reply:
Sorry, I wasn't clear. My suggestion is to close this PR and revise the [docs](https://mui.com/material-ui/customization/palette/#adding-new-colors) about adding new colors, so that it can be used in all components' color prop.

Because I think this PR does not fix the root cause. The root cause is the lack of documentation. We haven't mentioned that "if you want to add a palette that can be used in component color prop, here is how to do it".

So, my suggestion is to fix the docs instead of the implementation. Let me know, if this is still unclear.

Reply:
@siriwatknp The issue is about `contrastText` being passed into `alpha` function for "custom" defined colors due to which the component crashes. The documentation that you are suggesting to add a new palette is fine in the current docs. 

We also mention that if "contrastText" is omitted, its value will be calculated to contrast with "main". But for a "custom" palette's "main" color, the contrastText doesn't get calculated thus passing undefined to alpha.

So I believe that if it gets calculated for default primary, secondary palettes, it should for custom palettes as well using the `main` value.

Thus there is no need of revising the docs I think. Do read the PR's description. The user who reported the issue has a custom color palette which I believe was clear in the docs for him.

I hope it's now clear.

Reply:
> Can you point me to the docs that mention this?

Under [providing the colors directly](https://mui.com/material-ui/customization/palette/#providing-the-colors-directly) section.

Reply:
So looks like the only thing we need to update is the codesnippet in [Providing colors directly](https://mui.com/material-ui/customization/palette/#providing-the-colors-directly) to something like:

```diff --git a/docs/data/material/customization/palette/palette.md b/docs/data/material/customization/palette/palette.md
index 2eb22a39bb..32998ca3a2 100644
--- a/docs/data/material/customization/palette/palette.md
+++ b/docs/data/material/customization/palette/palette.md
@@ -87,6 +87,14 @@ const theme = createTheme({
       // dark: will be calculated from palette.secondary.main,
       contrastText: '#ffcc00',
     },
+    // All tokens need to be provided for custom colors if this
+    // color is going to be used as a prop in the Material UI's components
+    custom: {
+      light: '#ffa726'
+      main: '#f57c00',
+      dark: '#ef6c00',
+      contrastText: 'rgba(0, 0, 0, 0.87)',
+    }
     // Used by `getContrastText()` to maximize the contrast between
     // the background and the text.
     contrastThreshold: 3,
```